### PR TITLE
cgo,testlapack: add Dlacn2 and its test

### LINKED ISSUE #200: testlapack: missing test for Dlacn2

### PR DESCRIPTION
**Pull request by vladimir-ch**
Updates #142 
Resolves #200

### CODE REVIEW

File: cgo/lapack.go [hunk 1]
```diff
@@ -102,6 +102,59 @@ type Implementation struct{}
...
+// all other parameters should be unchanged during sequential calls, and the updated
+// values of est and kase should be used. On the final return (when kase is returned
+// as 0), V = A * W, where est = norm(V) / norm(W).
+//
+// isign, v, and x must all have length n and will panic otherwise. isave is used
+// for temporary storage.
+//
+// Dlacn2 is an internal routine. It is exported for testing purposes.
+func (impl Implementation) Dlacn2(n int, v, x []float64, isgn []int, est float64, kase int, isave *[3]int) (float64, int) {
+	if n < 1 {
```
Comment (vladimir-ch) [inline cgo/lapack.go]: For consistency this check should be added also to native.

Comment (vladimir-ch) [inline cgo/lapack.go]: ... and documented.

File: cgo/lapack.go [hunk 2]
```diff
@@ -102,6 +102,59 @@ type Implementation struct{}
...
+//  A^T * X if kase == 2
+// all other parameters should be unchanged during sequential calls, and the updated
+// values of est and kase should be used. On the final return (when kase is returned
+// as 0), V = A * W, where est = norm(V) / norm(W).
+//
+// isign, v, and x must all have length n and will panic otherwise. isave is used
+// for temporary storage.
+//
+// Dlacn2 is an internal routine. It is exported for testing purposes.
+func (impl Implementation) Dlacn2(n int, v, x []float64, isgn []int, est float64, kase int, isave *[3]int) (float64, int) {
```
Comment (vladimir-ch) [inline cgo/lapack.go]: I've been thinking about using `[]int` for isave and check its length but I'm not sure it's worth it.

File: cgo/lapack.go (hunk 1 above)
Comment (kortschak) [inline cgo/lapack.go]: File issue?

File: cgo/lapack.go [hunk 3]
```diff
@@ -102,6 +102,59 @@ type Implementation struct{}
...
+
+	isgn32 := make([]int32, n)
+	for i, v := range isgn {
+		isgn32[i] = int32(v)
+	}
+	pest := make([]float64, 1)
+	pest[0] = est
+	kase32 := make([]int32, 1)
+	kase32[0] = int32(kase)
+	isave32 := make([]int32, 3)
```
Comment (kortschak) [inline cgo/lapack.go]:
```
pest := []float64{est}
kase32 := []int32{int32(kase)}
isave32 := []int32{int32(isave[0]), int32(isave[1]), int32(isave[2])}
```

Could save an allocation at the cost of readability by putting kase32 and isave32 into the same slice.

**Pull request merged**